test/async-test.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+
+const fastify = require('fastify')()
```

Comment:
We could copy the same logic of the other test using the `const build = require('./example')` but we could create a new file `example-async.js` that is equal to `example` but use the promise instead with the same tests. It should be simple and so we can cover all the cases.

And of course, we should also test the mixed case (one promise and one sync function). The edge case 💪

Reply:
Agreed, I've based new tests on top of new `example-async.js` and updated both examples with explicit `reply` call from the auth function.